### PR TITLE
Feature: add repository-specific audit settings

### PR DESCRIPTION
**Pull request by austin-weeks**
- added `--settings` flag to audit command - allows passing a JSON string containing repository specific audit settings
    - JSON string is validated in `audit_settings.py` and appended to `Repository` objects
    - repository-specific CLI flags are now passed to RepoAuditor in `audit_repository` function
- added documentation and tests for above

### CODE REVIEW

File: tests/test_audit_settings.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        monkeypatch.setattr(
+            self._validate_settings_json_IMPORT_PATH,
+            lambda *args, **kwargs: SETTINGS_JSON,
+        )
+
+        repos_with_settings = append_settings(REPOS, "settings")
+        for repo in repos_with_settings:
+            assert repo.cli_flags is None
+
+    def test_added_settings(self, monkeypatch: MonkeyPatch):
```
Comment (brzdev) [inline tests/test_audit_settings.py:93]: Would be good to also test a case where a repo has an empty string value in settings (i.e. {"repo_1": ""}) to confirm expected behavior.

File: src/orgwarden/repository.py [hunk 2]
```diff
@@ -6,3 +6,4 @@ class Repository:
     name: str
     url: str
     org: str
+    cli_flags: str | None = None
```
Comment (brzdev) [inline src/orgwarden/repository.py:9]: Would be good to document this new attributes and current attributes in a class docstring so future contributors know its purpose and format expectations.

File: README.md [hunk 3]
```diff
@@ -92,6 +97,33 @@ repo - public_repo ✅
...
+```json
+{
+  "my-awesome-next-app": "--GitHub-AllowDeletions-true --GitHub-SupportWikis-false",
+  "TopSecretInternalTool": "--GitHub-AutoMerge-false"
+}
+```
+...OrgWarden will run RepoAuditor against the `my-awesome-next-app` repository with the `--GitHub-AllowDeletions-true` and `--GitHub-SupportWikis-false` flags, and against the `TopSecretInternalTool` repository with the `--GitHub-AutoMerge-false` flag.
+
+For more information on available CLI flags, please visit the [RepoAuditor docs](https://github.com/gt-sse-center/RepoAuditor).
+
```
Comment (brzdev) [inline README.md:124]: One thought, how does OrgWarden handle invalid or unrecognized flags per repo? Might be helpful to clarify whether those raise errors, warnings, or get silently ignored.

File: tests/test_audit_settings.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+            data = _validate_settings_json(json_string)
+            assert data == schema
+
+
+class TestAppendSettings:
+    _validate_settings_json_IMPORT_PATH = (
+        "orgwarden.audit_settings._validate_settings_json"
+    )
+
+    def test_validate_settings_json_called_correctly(self, monkeypatch: MonkeyPatch):
```
[the remark this answers is not in the document]
Comment (brzdev) [reply tests/test_audit_settings.py:59]: Agreed with the approach. Since `_validate_settings_json` is already tested independently, mocking it here avoids redundant logic checks and keeps the tests isolated. The lambda pattern is lightweight and readable too. No changes needed from me.

**Review by brzdev — APPROVED** [review]
Approved with a few minor comments, none are blocking though. Nice work!

File: src/orgwarden/audit_settings.py [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+from dataclasses import dataclass, replace
+
+
+@dataclass(frozen=True)
+class RepositorySettings:
+    repo_name: str
+    cli_flags: str
+
+
+# gingerly use of hellspawn
```
Comment (davidbrownell) [inline src/orgwarden/audit_settings.py:13]: What does this mean?

File: src/orgwarden/audit_settings.py [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+
+    for repo_settings in settings:
+        repo_name, cli_flags = repo_settings.repo_name, repo_settings.cli_flags
+        if repo_name in flags_dict:
+            raise ValueError(
+                f"Repository settings sequence contains multiple entries for {repo_name}."
+            )
+        flags_dict[repo_name] = cli_flags
+
+    copied_repos = [replace(repo) for repo in repos]
```
Comment (davidbrownell) [inline src/orgwarden/audit_settings.py:57]: Why are the repos being copied to a new list?

In trying to figure out the answer to this question, I see that the copy of the input repos are being returned, but I don't see anything that is actually modifying the repos list.

My hunch is that there is a cleaner way to accomplish what is being accomplished.

File: tests/test_audit_settings.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        new_repos = append_settings(REPOS, [])
+        for new_repo, og_repo in zip(new_repos, REPOS):
+            assert new_repo is not og_repo
+
+    def test_raises_with_duplicate_repos(self):
+        REPOSITORY_SETTINGS = [
+            RepositorySettings("repo", "value 1"),
+            RepositorySettings("repo", "value 2"),
+        ]
+        with pytest.raises(ValueError):
```
Comment (davidbrownell) [inline tests/test_audit_settings.py:66]: It's generally a good idea to test not only the type of exception, but also the text of the exception as well. This can be useful if the called function can raise `ValueError` for a variety of different reasons.

Consider:

```
with pytest.raises(ValueError, "I expect this string to be in the error message"):
```

**Review by davidbrownell — CHANGES_REQUESTED** [review]

**Review by davidbrownell — APPROVED** [review]
